### ISSUE TITLE
level-set 0.3.0 build failure under Octave.app 4.4.1-u1

### ISSUE THREAD

**Issue by apjanke** [octave-app/octave-app#158]
```
>> pkg install -forge level-set
2019-10-21 00:55 octave-gui[96168] (FSEvents.framework) FSEventStreamFlushSync(): failed assertion '(SInt64)last_id > 0LL'

2019-10-21 00:55 octave-gui[96168] (FSEvents.framework) FSEventStreamFlushSync(): failed assertion '(SInt64)last_id > 0LL'

In file included from FastMarching.cpp:19:
In file included from ./FastMarching.hpp:56:
./Heap.hpp:78:15: warning: deleted function definitions are a C++11 extension [-Wc++11-extensions]
    Heap () = delete;
              ^
./Heap.hpp:79:28: warning: deleted function definitions are a C++11 extension [-Wc++11-extensions]
    Heap (const Heap& o) = delete;
                           ^
./Heap.hpp:80:39: warning: deleted function definitions are a C++11 extension [-Wc++11-extensions]
    Heap& operator= (const Heap& o) = delete;
                                      ^
./Heap.hpp:205:20: warning: deleted function definitions are a C++11 extension [-Wc++11-extensions]
    EntryData () = delete;
                   ^
./Heap.hpp:206:38: warning: deleted function definitions are a C++11 extension [-Wc++11-extensions]
    EntryData (const EntryData& o) = delete;
                                     ^
./Heap.hpp:207:49: warning: deleted function definitions are a C++11 extension [-Wc++11-extensions]
    EntryData& operator= (const EntryData& o) = delete;
                                                ^
In file included from FastMarching.cpp:19:
In file included from ./FastMarching.hpp:56:
In file included from ./Heap.hpp:260:
./Heap.tpp:35:8: warning: 'auto' type specifier is a C++11 extension [-Wc++11-extensions]
  for (auto e : data)
       ^
./Heap.tpp:35:15: warning: range-based for loop is a C++11 extension [-Wc++11-extensions]
  for (auto e : data)
              ^
./Heap.tpp:169:13: warning: 'auto' type specifier is a C++11 extension [-Wc++11-extensions]
      const auto child = down (i);
            ^
./Heap.tpp:192:9: warning: 'auto' type specifier is a C++11 extension [-Wc++11-extensions]
  const auto indexA = a.index;
        ^
./Heap.tpp:193:9: warning: 'auto' type specifier is a C++11 extension [-Wc++11-extensions]
  const auto indexB = b.index;
        ^
./Heap.tpp:228:9: warning: 'auto' type specifier is a C++11 extension [-Wc++11-extensions]
  const auto ind = 2 * index + c + 1;
        ^
In file included from FastMarching.cpp:19:
./FastMarching.hpp:162:44: warning: deleted function definitions are a C++11 extension [-Wc++11-extensions]
  UpdateEquation (const UpdateEquation&) = delete;
                                           ^
./FastMarching.hpp:163:55: warning: deleted function definitions are a C++11 extension [-Wc++11-extensions]
  UpdateEquation& operator= (const UpdateEquation&) = delete;
                                                      ^
./FastMarching.hpp:241:17: warning: deleted function definitions are a C++11 extension [-Wc++11-extensions]
  EqnEntry () = delete;
                ^
./FastMarching.hpp:242:32: warning: deleted function definitions are a C++11 extension [-Wc++11-extensions]
  EqnEntry (const EqnEntry&) = delete;
                               ^
./FastMarching.hpp:243:43: warning: deleted function definitions are a C++11 extension [-Wc++11-extensions]
  EqnEntry& operator= (const EqnEntry&) = delete;
                                          ^
./FastMarching.hpp:347:14: warning: deleted function definitions are a C++11 extension [-Wc++11-extensions]
  Entry () = delete;
             ^
./FastMarching.hpp:348:28: warning: deleted function definitions are a C++11 extension [-Wc++11-extensions]
  Entry (const Entry& o) = delete;
                           ^
./FastMarching.hpp:349:39: warning: deleted function definitions are a C++11 extension [-Wc++11-extensions]
  Entry& operator= (const Entry& o) = delete;
                                      ^
./FastMarching.hpp:534:13: warning: deleted function definitions are a C++11 extension [-Wc++11-extensions]
  Grid () = delete;
            ^
./FastMarching.hpp:535:26: warning: deleted function definitions are a C++11 extension [-Wc++11-extensions]
  Grid (const Grid& o) = delete;
                         ^
./FastMarching.hpp:536:37: warning: deleted function definitions are a C++11 extension [-Wc++11-extensions]
  Grid& operator= (const Grid& o) = delete;
                                    ^
In file included from FastMarching.cpp:19:
In file included from ./FastMarching.hpp:614:
./FastMarching.tpp:69:13: warning: 'auto' type specifier is a C++11 extension [-Wc++11-extensions]
      const auto oldValue = neighbour[d];
            ^
FastMarching.cpp:40:8: warning: 'auto' type specifier is a C++11 extension [-Wc++11-extensions]
  for (auto e : entries)
       ^
FastMarching.cpp:40:15: warning: range-based for loop is a C++11 extension [-Wc++11-extensions]
  for (auto e : entries)
              ^
FastMarching.cpp:56:3: warning: 'auto' type specifier is a C++11 extension [-Wc++11-extensions]
  auto iter = entries.find (d);
  ^
FastMarching.cpp:81:14: warning: 'auto' type specifier is a C++11 extension [-Wc++11-extensions]
  for (const auto& e : entries)
             ^
FastMarching.cpp:81:22: warning: range-based for loop is a C++11 extension [-Wc++11-extensions]
  for (const auto& e : entries)
                     ^
FastMarching.cpp:105:18: warning: 'auto' type specifier is a C++11 extension [-Wc++11-extensions]
      for (const auto cur : sortedEntries)
                 ^
FastMarching.cpp:105:27: warning: range-based for loop is a C++11 extension [-Wc++11-extensions]
      for (const auto cur : sortedEntries)
                          ^
FastMarching.cpp:117:7: error: no matching function for call to 'issueWarning'
      issueWarning ("level-set:fast-marching:too-far-alive",
      ^~~~~~~~~~~~
./FastMarching.hpp:102:6: note: candidate function not viable: no known conversion from 'const char [38]' to 'const std::string' (aka
      'const basic_string<char, char_traits<char>, allocator<char> >') for 1st argument
void issueWarning (const std::string& id, const std::string& fmt, ...);
     ^
FastMarching.cpp:156:14: warning: 'auto' type specifier is a C++11 extension [-Wc++11-extensions]
  for (const auto cur : sortedEntries)
             ^
FastMarching.cpp:156:23: warning: range-based for loop is a C++11 extension [-Wc++11-extensions]
  for (const auto cur : sortedEntries)
                      ^
FastMarching.cpp:220:9: warning: 'auto' type specifier is a C++11 extension [-Wc++11-extensions]
  const auto buildEqn
        ^
FastMarching.cpp:221:7: error: expected expression
    = [&eqn, &grid] (const IndexTuple& neighbour, dimensionT d, indexT)
      ^
FastMarching.cpp:240:9: error: no matching function for call to 'issueWarning'
        issueWarning ("level-set:fast-marching:increased-distance",
        ^~~~~~~~~~~~
./FastMarching.hpp:102:6: note: candidate function not viable: no known conversion from 'const char [43]' to 'const std::string' (aka
      'const basic_string<char, char_traits<char>, allocator<char> >') for 1st argument
void issueWarning (const std::string& id, const std::string& fmt, ...);
     ^
FastMarching.cpp:305:9: warning: 'auto' type specifier is a C++11 extension [-Wc++11-extensions]
  const auto updateIt
        ^
FastMarching.cpp:306:7: error: expected expression
    = [this] (const IndexTuple& neighbour, dimensionT, indexT)
      ^
FastMarching.cpp:345:8: warning: 'auto' type specifier is a C++11 extension [-Wc++11-extensions]
  for (auto i : s)
       ^
FastMarching.cpp:345:15: warning: range-based for loop is a C++11 extension [-Wc++11-extensions]
  for (auto i : s)
              ^
FastMarching.cpp:349:8: warning: 'auto' type specifier is a C++11 extension [-Wc++11-extensions]
  for (auto& p : entries)
       ^
FastMarching.cpp:349:16: warning: range-based for loop is a C++11 extension [-Wc++11-extensions]
  for (auto& p : entries)
               ^
FastMarching.cpp:363:8: warning: 'auto' type specifier is a C++11 extension [-Wc++11-extensions]
  for (auto p : entries)
       ^
FastMarching.cpp:363:15: warning: range-based for loop is a C++11 extension [-Wc++11-extensions]
  for (auto p : entries)
              ^
FastMarching.cpp:407:14: warning: 'auto' type specifier is a C++11 extension [-Wc++11-extensions]
  for (const auto center : entries)
             ^
FastMarching.cpp:407:26: warning: range-based for loop is a C++11 extension [-Wc++11-extensions]
  for (const auto center : entries)
                         ^
43 warnings and 4 errors generated.
make: *** [FastMarching.o] Error 1
make: *** Waiting for unfinished jobs....
Utils.cpp:36:19: warning: passing an object of reference type to 'va_start' has undefined behavior [-Wvarargs]
  va_start (args, fmt);
                  ^
Utils.cpp:33:57: note: parameter of type 'const std::string &' (aka 'const basic_string<char, char_traits<char>, allocator<char> > &') is
      declared here
issueWarning (const std::string& id, const std::string& fmt, ...)
                                                        ^
1 warning generated.
c++ -g -O2 -Wall -Wextra -Weffc++ -pedantic -fPIC -c FastMarching.cpp -o FastMarching.o
CXXFLAGS="-g -O2 -Wall -Wextra" /Applications/Octave-4.4.1.app/Contents/Resources/usr/Cellar/octave-octave-app@4.4.1/4.4.1/bin/mkoctfile-4.4.1 -c Utils.cpp -o Utils.o

pkg: error running `make' for the level-set package.
error: called from
    configure_make at line 99 column 9
    install at line 184 column 7
    pkg at line 437 column 9
>>
```

**Comment by apjanke** [comment]
This was already reported upstream: https://savannah.gnu.org/bugs/index.php?49294

**Comment by apjanke** [comment]
Same failure in vanilla Homebrewed Octave. Closing as Upstream/Wontfix.

On macOS 14.5 Intel:

In Octave.app 9.2.0 beta1:

```
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/locale:3867:28: note:
      'wbuffer_convert<octave::string::deletable_facet<octave::string::codecvt_u8>, char>' has been explicitly marked deprecated here
class _LIBCPP_TEMPLATE_VIS _LIBCPP_DEPRECATED_IN_CXX17 wbuffer_convert
                           ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__config:1037:41: note: expanded from macro
      '_LIBCPP_DEPRECATED_IN_CXX17'
#    define _LIBCPP_DEPRECATED_IN_CXX17 _LIBCPP_DEPRECATED
                                        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__config:1010:49: note: expanded from macro
      '_LIBCPP_DEPRECATED'
#      define _LIBCPP_DEPRECATED __attribute__((__deprecated__))
                                                ^
In file included from Utils.cpp:19:
In file included from ./Utils.hpp:26:
In file included from /Applications/Octave-9.2.0_beta1.app/Contents/Resources/usr/Cellar/octave-octapp@9.2.0/9.2.0/include/octave-9.2.0/octave/../octave/oct.h:35:
In file included from /Applications/Octave-9.2.0_beta1.app/Contents/Resources/usr/Cellar/octave-octapp@9.2.0/9.2.0/include/octave-9.2.0/octave/defun-dld.h:35:
In file included from /Applications/Octave-9.2.0_beta1.app/Contents/Resources/usr/Cellar/octave-octapp@9.2.0/9.2.0/include/octave-9.2.0/octave/defun-int.h:33:
In file included from /Applications/Octave-9.2.0_beta1.app/Contents/Resources/usr/Cellar/octave-octapp@9.2.0/9.2.0/include/octave-9.2.0/octave/ov-builtin.h:35:
In file included from /Applications/Octave-9.2.0_beta1.app/Contents/Resources/usr/Cellar/octave-octapp@9.2.0/9.2.0/include/octave-9.2.0/octave/ov-fcn.h:36:
In file included from /Applications/Octave-9.2.0_beta1.app/Contents/Resources/usr/Cellar/octave-octapp@9.2.0/9.2.0/include/octave-9.2.0/octave/ovl.h:38:
In file included from /Applications/Octave-9.2.0_beta1.app/Contents/Resources/usr/Cellar/octave-octapp@9.2.0/9.2.0/include/octave-9.2.0/octave/ov.h:69:
/Applications/Octave-9.2.0_beta1.app/Contents/Resources/usr/Cellar/octave-octapp@9.2.0/9.2.0/include/octave-9.2.0/octave/oct-stream.h:136:23: warning:
      'wbuffer_convert<octave::string::deletable_facet<octave::string::codecvt_u8>, char>' is deprecated [-Wdeprecated-declarations]
            (new std::wbuffer_convert<convfacet_u8, char>
                      ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/locale:3867:28: note:
      'wbuffer_convert<octave::string::deletable_facet<octave::string::codecvt_u8>, char>' has been explicitly marked deprecated here
class _LIBCPP_TEMPLATE_VIS _LIBCPP_DEPRECATED_IN_CXX17 wbuffer_convert
                           ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__config:1037:41: note: expanded from macro
      '_LIBCPP_DEPRECATED_IN_CXX17'
#    define _LIBCPP_DEPRECATED_IN_CXX17 _LIBCPP_DEPRECATED
                                        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__config:1010:49: note: expanded from macro
      '_LIBCPP_DEPRECATED'
#      define _LIBCPP_DEPRECATED __attribute__((__deprecated__))
                                                ^
Utils.cpp:36:19: warning: passing an object of reference type to 'va_start' has undefined behavior [-Wvarargs]
  va_start (args, fmt);
                  ^
Utils.cpp:33:57: note: parameter of type 'const std::string &' (aka 'const basic_string<char> &') is declared here
issueWarning (const std::string& id, const std::string& fmt, ...)
                                                        ^
4 warnings generated.
c++ -g -O2 -Wall -Wextra -Weffc++ -pedantic -fPIC -c FastMarching.cpp -o FastMarching.o
CXXFLAGS="-g -O2 -Wall -Wextra" /Applications/Octave-9.2.0_beta1.app/Contents/Resources/usr/Cellar/octave-octapp@9.2.0/9.2.0/bin/mkoctfile-9.2.0 -c Utils.cpp -o Utils.o

error: pkg: error running 'make' for the level-set package
error: called from
    configure_make at line 117 column 9
    install at line 202 column 7
    pkg at line 619 column 9
>>
```

In vanilla brewed Octave 9.2.0:

```
In file included from /usr/local/Cellar/octave/9.2.0/include/octave-9.2.0/octave/ov.h:69:
/usr/local/Cellar/octave/9.2.0/include/octave-9.2.0/octave/oct-stream.h:135:34: warning:
      'wbuffer_convert<octave::string::deletable_facet<octave::string::codecvt_u8>, char>' is deprecated [-Wdeprecated-declarations]
          = std::unique_ptr<std::wbuffer_convert<convfacet_u8, char>>
                                 ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/locale:3867:28: note:
      'wbuffer_convert<octave::string::deletable_facet<octave::string::codecvt_u8>, char>' has been explicitly marked deprecated here
class _LIBCPP_TEMPLATE_VIS _LIBCPP_DEPRECATED_IN_CXX17 wbuffer_convert
                           ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__config:1037:41: note:
      expanded from macro '_LIBCPP_DEPRECATED_IN_CXX17'
#    define _LIBCPP_DEPRECATED_IN_CXX17 _LIBCPP_DEPRECATED
                                        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__config:1010:49: note:
      expanded from macro '_LIBCPP_DEPRECATED'
#      define _LIBCPP_DEPRECATED __attribute__((__deprecated__))
                                                ^
In file included from Utils.cpp:19:
In file included from ./Utils.hpp:26:
In file included from /usr/local/Cellar/octave/9.2.0/include/octave-9.2.0/octave/../octave/oct.h:35:
In file included from /usr/local/Cellar/octave/9.2.0/include/octave-9.2.0/octave/defun-dld.h:35:
In file included from /usr/local/Cellar/octave/9.2.0/include/octave-9.2.0/octave/defun-int.h:33:
In file included from /usr/local/Cellar/octave/9.2.0/include/octave-9.2.0/octave/ov-builtin.h:35:
In file included from /usr/local/Cellar/octave/9.2.0/include/octave-9.2.0/octave/ov-fcn.h:36:
In file included from /usr/local/Cellar/octave/9.2.0/include/octave-9.2.0/octave/ovl.h:38:
In file included from /usr/local/Cellar/octave/9.2.0/include/octave-9.2.0/octave/ov.h:69:
/usr/local/Cellar/octave/9.2.0/include/octave-9.2.0/octave/oct-stream.h:136:23: warning:
      'wbuffer_convert<octave::string::deletable_facet<octave::string::codecvt_u8>, char>' is deprecated [-Wdeprecated-declarations]
            (new std::wbuffer_convert<convfacet_u8, char>
                      ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/locale:3867:28: note:
      'wbuffer_convert<octave::string::deletable_facet<octave::string::codecvt_u8>, char>' has been explicitly marked deprecated here
class _LIBCPP_TEMPLATE_VIS _LIBCPP_DEPRECATED_IN_CXX17 wbuffer_convert
                           ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__config:1037:41: note:
      expanded from macro '_LIBCPP_DEPRECATED_IN_CXX17'
#    define _LIBCPP_DEPRECATED_IN_CXX17 _LIBCPP_DEPRECATED
                                        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__config:1010:49: note:
      expanded from macro '_LIBCPP_DEPRECATED'
#      define _LIBCPP_DEPRECATED __attribute__((__deprecated__))
                                                ^
Utils.cpp:36:19: warning: passing an object of reference type to 'va_start' has undefined behavior [-Wvarargs]
  va_start (args, fmt);
                  ^
Utils.cpp:33:57: note: parameter of type 'const std::string &' (aka 'const basic_string<char> &') is declared here
issueWarning (const std::string& id, const std::string& fmt, ...)
                                                        ^
4 warnings generated.
c++ -g -O2 -Wall -Wextra -Weffc++ -pedantic -fPIC -c FastMarching.cpp -o FastMarching.o
CXXFLAGS="-g -O2 -Wall -Wextra" /usr/local/Cellar/octave/9.2.0/bin/mkoctfile-9.2.0 -c Utils.cpp -o Utils.o

error: pkg: error running 'make' for the level-set package
error: called from
    configure_make at line 117 column 9
    install at line 202 column 7
    pkg at line 619 column 9
octave:26>
```
